tycho-indexer/src/extractor/reorg_buffer.rs

```diff
@@ -105,7 +106,12 @@ where
         &mut self,
         final_block_height: u64,
     ) -> Result<Vec<B>, StorageError> {
-        let target_index = self.find_index(|b| b.block().number == final_block_height);
+        let target_index = if self.last_finalized_block.replace(final_block_height) != Some(final_block_height) {
+            self.find_index(|b| b.block().number == final_block_height).map(|idx| idx + 1)
```

Comment:
Isn't this equivalent and more concise? (under the assumption that blocks are ordered and without gaps)
```suggestion
            self.find_index(|b| b.block().number > final_block_height)
```